### PR TITLE
Handle indexing of *hasMany-through* associations.

### PR DESCRIPTION
**Pull request by mphasize**
This PR enables the `Ember.buildResponse` function to handle Many-to-Many-Through associations as currently supported in Sails.

### CODE REVIEW

**Comment by mphasize** [comment]
See https://gist.github.com/mphasize/aeaff696132357797e0b on how to setup Through associations.

**Comment by Globegitter** [comment]
@mphasize awesome example - since when does that exist? Is not even stated in the documentation yet: http://sailsjs.org/#!/documentation/concepts/ORM/Associations/ThroughAssociations.html

**Comment by mphasize** [comment]
@Globegitter As you can see in the Gist I put it together from scraps and StackOverflow comments on May 30, 2014. ;-) I am using this extensively, mostly to sanitize the join-table naming in my database.

I think I also got a few issues with Waterline fixed to make sure that this works and so far, it does. :-)

**Comment by Globegitter** [comment]
@mphasize Aahh, that is good to know. Just putting your example in a PR for the docs. Could you maybe explain it in plain english? To be precise, what is the `via: 'app', groupBy: 'user'` doing in the junction table?

I have to say I am glad we have working relationships but the syntax is constantly confusing me, what exactly which keyword means and when to put it where. The ED syntax feels much easier to me on this one. Anyway, just a comment :P

**Comment by mphasize** [comment]
I'm afraid I don't understand all the properties myself. Like I said, I put it together from StackOverflow comments similar to these ( http://stackoverflow.com/questions/27013077/sails-js-waterline-many-to-many-through-association, I can't find the one were somebody from Sails answered ) and looking at the source code of automatic join table definitions in Sails / Waterline.

My best guess is that the properties simply replicate keys and input needed to build appropriate queries. Think about MySQL join queries and `ON criteria`...
